core/AssetManager.php

```diff
@@ -138,20 +138,21 @@ public function getCssInclusionDirective()
     /**
      * Return JS file inclusion directive(s) using the markup <script>
      *
+     * @param bool $deferJS
      * @return string
      */
-    public function getJsInclusionDirective()
+    public function getJsInclusionDirective(bool $deferJS = false)
     {
         $result = "<script type=\"text/javascript\">\n" . StaticContainer::get('Piwik\Translation\Translator')->getJavascriptTranslations() . "\n</script>";
 
         if ($this->isMergedAssetsDisabled()) {
             $this->getMergedCoreJSAsset()->delete();
             $this->getMergedNonCoreJSAsset()->delete();
 
-            $result .= $this->getIndividualCoreAndNonCoreJsIncludes();
+            $result .= $this->getIndividualCoreAndNonCoreJsIncludes($deferJS);
```

Comment:
Guess we also need to set `deferJS` to `false` here. When merged assets are disabled we can't server the javascript assets defered. Problem is that they might get loaded in incorrect order that way. At least I'm seeing javascript errors having merged assets disabled.

Reply:
I guess we can remove it here for now but if the order matters in how they are merged or how they are served unmerged, we probably should look into the whole system. But that the Matomo 6 thing.

Ok, I'll remove it here. Not just set to false, we can remove the whole mechanism here and in the nested calls as that's the only place where it's used.

Reply:
Our system ensure that certain scripts are loaded first, like jQuery or vue.js. When assets are merged they are placed first in the files. But when using unmerged assets another assets might get loaded faster, running into problems as jquery or vue might not be loaded yet.
As using unmerged assets is actually only used for development there shouldn't be any need to change anything around that.